financial-connections/src/main/java/com/stripe/android/financialconnections/repository/FinancialConnectionsRepository.kt

```diff
@@ -135,26 +138,38 @@ internal class FinancialConnectionsRepositoryImpl @Inject constructor(
 
     override suspend fun createPaymentMethod(
         paymentDetailsId: String,
-        consumerSessionClientSecret: String
+        consumerSessionClientSecret: String,
+        billingAddress: BillingAddress?,
+        billingEmailAddress: String,
     ): PaymentMethod {
-        val credentials = mapOf(
-            "consumer_session_client_secret" to consumerSessionClientSecret,
-        )
-
-        val params = mapOf(
+        val linkParams = mapOf(
             "type" to "link",
             "link" to mapOf(
-                "credentials" to credentials,
+                "credentials" to mapOf(
+                    "consumer_session_client_secret" to consumerSessionClientSecret,
+                ),
                 "payment_details_id" to paymentDetailsId,
             ),
         )
 
+        val billingDetails = buildMap {
+            if (billingAddress != null) {
+                putAll(billingAddress.apiParams())
+            }
```

Comment:
nit - the `let` notation you used on other places looks better : )

Reply:
just making sure I understand:

- `billingAddress.apiParams()` is the mapping for the payment method endpoint
- `billingAddress.consumerApiParams` is used on consumer endpoints, like create payment details.

I can see some confusion if that's the case, since one is an inner method within the model and the other is an extension function.

Maybe we can move both to a `mappers` file with some comments on how to use them?